packages/flutter/lib/src/widgets/system_context_menu.dart

```diff
@@ -63,15 +70,26 @@ class SystemContextMenu extends StatefulWidget {
           editableTextState.textEditingValue.selection,
         ),
       ),
-      onSystemHide: () {
-        editableTextState.hideToolbar();
-      },
+      items: items,
+      onSystemHide: editableTextState.hideToolbar,
     );
   }
 
   /// The [Rect] that the context menu should point to.
   final Rect anchor;
 
+  /// A list of the items to be displayed in the system context menu.
+  ///
+  /// If none are given, the items will be inferred by the platform based on the
+  /// current [TextInputConnection].
+  ///
+  /// Built-in items will only be shown when relevant. For example, if
+  /// [IOSSystemContextMenuItemCopy] is passed, the copy button will only be
+  /// shown when there is a non-empty selection and not when the selection is
+  /// collapsed. It's not necessary to manually add and remove these items based
+  /// on the state of the field.
+  final List<IOSSystemContextMenuItem>? items;
```

Comment:
Should `IOSSystemContextMenuItem` have a more generic base class `SystemContextMenuItem`? I mostly ask because while `SystemContextMenu` is only supported on iOS right now it does have a generic name so maybe we should follow that pattern here.

Reply:
I think if I did this again from scratch I would have named SystemContextMenu to IOSSystemContextMenu... Lately I've been trying to match the platform APIs in Dart and then abstract them later.

My main concern with dropping the IOS prefix is that we'll implement this for Android and then realize that the API isn't quite compatible, and SystemContextMenuItem needs to be deprecated, or I need to come up with a different generic name. However with the prefix, I have the freedom to make a parallel AndroidSystemContextMenuItem, and/or a SystemContextMenuItem that covers both.

Do you think that's worth it or is it too confusing?

Reply:
Yeah I see the concern here, i'm not sure what's best here but I will throw another option out there: what about an API like `iOSMenuItems, androidMenuItems, windowsMenuItems, linuxMenuItems`? That way each "*SystemContextMenuItem" can adhere to its specific platform API without conflicts, and we can always try to move towards a single "items" API in the future.